packages/playwright-core/src/server/fetch.ts

```diff
@@ -307,8 +309,27 @@ export abstract class APIRequestContext extends SdkObject {
       // If we have a proxy agent already, do not override it.
       const agent = options.agent || (url.protocol === 'https:' ? httpsHappyEyeballsAgent : httpHappyEyeballsAgent);
       const requestOptions = { ...options, agent };
+
+      const startAt = monotonicTime();
+      let dnsLookupAt: number | undefined;
+      let tcpConnectionAt: number | undefined;
+      let tlsHandshakeAt: number | undefined;
+      let requestFinishAt: number | undefined;
+      let firstByteAt: number | undefined;
+      let endAt: number | undefined;
+
       const request = requestConstructor(url, requestOptions as any, async response => {
         const notifyRequestFinished = (body?: Buffer) => {
+          const timings: har.Timings = {
+            send: requestFinishAt! - startAt,
+            wait: firstByteAt! - requestFinishAt!,
+            receive: endAt! - firstByteAt!,
+            dns: dnsLookupAt ? dnsLookupAt - startAt : -1,
+            connect: (tlsHandshakeAt ?? tcpConnectionAt!) - startAt,
```

Comment:
Why not just `tcpConnectionAt - startAt`? This warrants a comment at least.

Reply:
fair! added comments in 947942349aa4c08cb7942821ad4bf09984d0b332